integration/solana/simple_collectible.spec.ts

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: Apache-2.0
+
+// DISCLAIMER: This file is an example of how to mint and transfer NFTs on Solana. It is not production ready and has not been audited for security.
+// Use it at your own risk.
+
+import {loadContract} from "./setup";
+import {Keypair, PublicKey, SystemProgram} from "@solana/web3.js";
+import {publicKeyToHex, HexToPublicKey} from "@solana/solidity";
+import {createMint, getOrCreateAssociatedTokenAccount, TOKEN_PROGRAM_ID} from "@solana/spl-token";
+
+describe('Simple collectible', function() {
+    this.timeout(500000);
+
+
+    it('nft example', async function mint_nft() {
+        const mint_authority = Keypair.generate();
+        const { contract, connection, payer, program } = await loadContract('SimpleCollectible', 'SimpleCollectible.abi', [publicKeyToHex(mint_authority.publicKey)]);
+
+        // Save the payer in Solidity
+        await contract.functions.set_payer(
+            publicKeyToHex(payer.publicKey)
+        );
+
+        const freezeAuthority = Keypair.generate();
+        // Create and initialize a new mint based on the funding account and a mint authority
+        const mint = await createMint(
+            connection,
+            payer,
+            mint_authority.publicKey,
+            freezeAuthority.publicKey,
+            3
+        );
+
+        // On Solana, an account must have an associated token account to save information about how many tokens
+        // the owner account owns. The associated account depends on both the mint account and the owner
+        const owner = await getOrCreateAssociatedTokenAccount(
+            connection,
+            payer,
+            mint, // Mint account
+            payer.publicKey // Owner account
+        );
+
+        // Create a collectible for an owner given a mint account. Each new NFT requires a new mint account.
+        const nft_id = await contract.functions.createCollectible(
+            "www.nft.com",
+            publicKeyToHex(mint),
+            publicKeyToHex(owner.address),
+            {
+                accounts: [TOKEN_PROGRAM_ID],
+                writableAccounts: [mint, owner.address],
+                signers: [mint_authority]
+            }
+        );
+
+        // Retrieve the owner of an NFT
+        const response = await contract.functions.get_nft_owner(
+            nft_id.result
+        );
```

Comment:
no check that the return values make sense

Reply:
If there was no error regenerating the public key for them, I assume they do make sense. What do you suggest here?

Reply:
Oh, I haven't verified if the nft_id exists in the mapping.